os/kernel/binary_manager/binary_manager_load.c

```diff
@@ -580,47 +598,128 @@ void binary_manager_release_binary_sem(int bin_idx)
 	irqrestore(flags);
 }
 
+#ifdef CONFIG_SUPPORT_COMMON_BINARY
 /****************************************************************************
- * Name: binary_manager_read_header
+ * Name: binary_manager_read_common_header
  *
  * Description:
- *	 This function reads header and checks whether it is valid or not.
+ *	 This function reads common binary header and checks whether it is valid or not.
  *
  ****************************************************************************/
-int binary_manager_read_header(char *path, user_binary_header_t *header_data, bool crc_check)
+int binary_manager_read_common_header(char *devpath, common_binary_header_t *header_data, bool crc_check)
```

Comment:
This function is almost same as `binary_manager_read_user_header`.
I think we can merge them.
Then we can reduce the codes.

Reply:
Right. I merged them, please refer the `binary_manager_read_header` function,